packages/docusaurus-utils/src/platformDependant.ts

```diff
@@ -0,0 +1,8 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+export type PlatformDependant<Type> = Type | Record<string, Type>;
```

Comment:
I don't see a need to create a file for this... `T | Record<string, T>` is not a very tedious type. Also, move this to `theme-common` because utils is for Node code, not client code.